File: UI/Renty/src/app/types/product/ProductDetailDto.ts

import { CategoryDto } from "../category/CategoryDto";
import { ImageDto } from "../image/ImageDto";
import { RatingDto } from "../rating/RatingDto";
import { UserDetailDto } from "../user/UserDetailDto";

export interface ProductDetailDto {
  id: number;
  name: string;
  price: number;
  category: CategoryDto | null;
  images: ImageDto[] | null;
  ratings: RatingDto[] | null;
  description: string;
  status: string;
  user: UserDetailDto | null;
  addingDate: string;
}

File: APIs/Renty/Application/DTOs/Product/ProductDetailDto.cs
namespace Renty.Application.DTOs
{

  public class ProductDetailDto
  {
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public CategoryDto? Category { get; set; }
    public List<ImageDto>? Images { get; set; }
    public List<RatingDto>? Ratings { get; set; }
    public string Description { get; set; } = string.Empty;
    public string Status { get; set; } = string.Empty;
    public UserDetailDto? User { get; set; }
    public DateTime AddingDate { get; set; }
  }

}